Controllers/PostController.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Blogger.Data;
+using Blogger.Data.Dto;
+using Blogger.Services;
+
+namespace Blogger.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : ControllerBase
+    {
+
+        private readonly PostService _PostService;
+
+        public PostController(PostService postService)
+        {
+            _PostService = postService;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> CreatePost([FromBody] PostDto formData)
+        {
+            if(formData.IsValid())
+            {
+                var result = await _PostService.Create(formData.GetObject());
+                return Created(result.Id.ToString(), result);
```

Comment:
What will happen if the create fails here for some reason? What the service will return? 
What would be the response that user will get if the insertion fails for some reason into the database?

Reply:
this will arise a **bug** 🐛

Reply:
How about using custom exception handler middleware
```c#
app.UseExceptionHandler(a => a.Run(async context =>
{
      var exception = context.Features.Get<IExceptionHandlerPathFeature>();
      var result = JsonSerializer.Serialize(new {error = exception.Error.Message});
      context.Response.ContentType = "application/json";
      await context.Response.WriteAsync(result);
}));
```

And I got this response when creation object instance is `null`
```json
{
  "error": "Object reference not set to an instance of an object."
}
```
When Database connection failed
```json
{
  "error": "A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 40 - Could not open a connection to SQL Server)"
}
```

header
```
 cache-control: no-cache 
 content-type: application/json 
 date: Wed, 26 Feb 2020 04:07:11 GMT 
 expires: -1 
 pragma: no-cache 
 server: Kestrel 
 status: 500 
```